Automattic/wp-calypso
# Staging Sites: Fix incorrect plan display under sites page

Fixes https://github.com/Automattic/dotcom-forge/issues/1953


## Proposed Changes

In this pr, I propose fixing the staging incorrect plan display under the sites page, by using the plan from the associated production site.

**Some notes for this pr:**

- Initially, I tried to implement this on the server side. This had some problems, as shown [here](p1679310173536649-slack-C04GESRBWKW)
- Under the `/sites` page we use the `useQuery` hook. Using this hook, the fetched doesn't end up in the main reducer that the site uses (`client/state/sites/reducer.js`). We end up fetching the sites twice. **We may need to revisit that.** I created [this issue ](https://github.com/Automattic/wp-calypso/issues/74647) for  having further discussions.
- As a result, this introduced some discrepancies. `useQuery` was fetching `wpcom_production_blog_id` and `wpcom_staging_blog_ids` but the "main" reducer didn't. I added those fields there as well. Also, please note, that I didn't handle the plans in the "main" reducer as well for safety reasons. `/sites` is a "self-contained" page. But changing the plans in the main reducer that is used everywhere, may introduce bugs, and will make things harder to test for this relatively simple pr.



## Testing Instructions

1. Make sure you have a staging site. If not you have to create one.
2. Visit `/sites`
3. Ensure that your staging site has the same plan as it's production counterpart.

![stage_prod_same_plan](https://user-images.githubusercontent.com/497103/227246179-67612770-1b18-4f49-8741-a32a5c40bcd3.png)




*

## Pre-merge Checklist
